docs/en/advanced/Scalability-Tuning.md

```diff
@@ -5,4 +5,84 @@ nickname: Scalability Tuning
 group: Advanced
 priority: 0
 ---
-TODO(adit): Add scalability related tuning tips
+
+## Introduction
+
+Alluxio is a scalable distributed file system designed to handle many workers within a single cluster.
+Several parameters can be tuned to prevent the Alluxio master from getting overloaded. This page
+aims to enumerate a list of such parameters which should be tuned for the scale of your cluster.
+
+### Alluxio Master Configuration
+
+#### Heap Size
+
+The Alluxio master heap size directly controls the total number of files that can fit into the
+master memory. Control the heap size by setting JVM options for the Alluxio master and secondary
+master processes. For example, to set the heap size to `256GB`, modify `alluxio-env.sh` as follows:
```

Comment:
Should `-Xms` also be set? I think it should be at least half to avoid unnecessary GC from growing the heap constantly.